src/main/java/core/basesyntax/dao/impl/StorageDaoImpl.java

```
@@ -0,0 +1,28 @@
+package core.basesyntax.dao.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.db.Storage;
+import java.util.Map;
+
+public class StorageDaoImpl implements StorageDao {
+
+    @Override
+    public void addFruitQuantity(String fruit, int quantity) {
+        Storage.putFruitQuantity(fruit, getFruitQuantity(fruit) + quantity);
```

addFruitQuantity directly uses getFruitQuantity(fruit) + quantity without validating inputs. The project requirements explicitly call out handling invalid inputs (for example: "Null parameters" and that negative quantities are invalid); you must validate method arguments and throw informative RuntimeException on invalid input (null/empty `fruit`, negative `quantity`) rather than allowing potentially corrupt state. Quote from requirements: "Remember that your code will need to be tested, so try to anticipate and handle all invalid input data in advance. For example: ... 3. Null parameters" and "Quantities must be integers; negative quantities are invalid for input; throw RuntimeException." Update this method to validate and fail fast with clear messages .